vars/deployPlannerSnapshot.groovy

```diff
@@ -0,0 +1,65 @@
+#!/usr/bin/groovy
+
+import java.util.regex.Pattern
+
+def call(body) {
+    // evaluate the body block, and collect configuration into the object
+    def config = [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = config
+    body()
+
+    def openShiftTemplate = config.openShiftTemplate
+    def originalImageName = config.originalImageName
+    def newImageName = config.newImageName
+    def deploymentName = config.githubRepo
+    def providerLabel = config.providerLabel ?: 'fabric8'
+    def project = config.githubProject
+
+    def flow = new io.fabric8.Fabric8Commands()
+    def utils = new io.fabric8.Utils()
+    def openShiftProject = config.openShiftProject + '-' + utils.getRepoName()
+
+    if (!flow.isAuthorCollaborator("", project)){
+        currentBuild.result = 'ABORTED'
+        error 'Change author is not a collaborator on the project, aborting build until we support the [test] comment'
+    }
+    def yaml = flow.getUrlAsString(openShiftTemplate)
+    def originalImage = "- image: ${originalImageName}:(.*)"
+    def newImage = "- image: ${newImageName}"
+    def compiledYaml = Pattern.compile(originalImage).matcher(yaml).replaceFirst(newImage)
+
```

Comment:
@pranavgore09 Its slightly odd that we use string matching and regular expressions to get data out of a structured data interchange format. I would have expected to see something like this.

```
def yaml = Yaml.parse(flow.getUrlAsString(openShiftTemplateURL))
yaml.services[1].image = config.newImage  // Whatever the real path is
def snapshot = yaml.toString()
// Remove the check below, because that wont fail now. 
```

I'm OK with this code, this might be written this way because of other nuances like groovy sandbox or something which we don't know. 

Point being, this feels like a very poor groovy implementation of 

```
wget openShiftTemplate > snapshot.yml
sed -i -e 's/oldImage/newImage/g' snapshot.yml
```
and that feels really odd.

Reply:
It indeed has a method to [parse yaml][read-yaml] but @jaseemabid mentioned I feel this is consistent with the existing implementation so lets go ahead with this and fix both later as a separate issue.

[read-yaml]: https://jenkins.io/doc/pipeline/steps/pipeline-utility-steps/#code-readyaml-code-read-yaml-from-files-in-the-workspace-or-text